src/Database/Type/DateTimeType.php

```diff
@@ -53,9 +53,12 @@ class DateTimeType extends Type implements TypeInterface
     /**
      * String format to use for DateTime parsing
      *
-     * @var string
+     * @var string|array
      */
-    protected $_format = 'Y-m-d H:i:s';
+    protected $_format = [
+        'Y-m-d H:i:s',
+        'Y-m-d\TH:i:s.uO',
```

Comment:
My concern is more that this adds a new valid input form that was not previously accepted. Developers may be surprised when previously invalid data becomes valid. Also concerning is that the new format includes timezone information which will require apps to ensure they are doing timezone conversions.